### PR TITLE
fix: invalid options on cardano-cli babbage transaction build and build-raw

### LINKED ISSUE #502: cardano-cli babbage transaction build AND build-raw show invalid options
cardano-cli 8.16.0.1 - linux-x86_64 - ghc-8.10
git rev 7f7a93d248b380d3dbff8c003a3ad66b6d1c770f

The flags  `--vote-file` and `--proposal-file` should not be present on any of 

* `cardano-cli babbage transaction build` 
* `cardano-cli babbage transaction build-raw`

the same occurs with *Legacy Commands*, also show the flags `--vote-file` and `--proposal-file` 

* `cardano-cli  transaction build` 
* `cardano-cli  transaction build-raw`

Of of the goals with having babbage as a top level command is be able to only present to the user flags that are relevant to that era, so these flag should not show under babbage.

### PR DESCRIPTION
**Pull request by felix-lipski**
# Changelog

```yaml
- description: |
    split eras in transaction build
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

[cardano-cli babbage transaction build AND build-raw show invalid options](https://github.com/input-output-hk/cardano-cli/issues/502)

# How to trust this PR

Run `cardano-cli <pre-conway> transaction build/build-raw` - it now doesn't show the `--vote-file` and `--proposal-file` flags.
Run `cardano-cli conway transaction build/build-raw` - it still shows the `--vote-file` and `--proposal-file` flags.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

**Comment by felix-lipski** [comment]
As per: https://github.com/input-output-hk/cardano-cli/issues/504#issuecomment-1845038871, I have only modified the era based commands, and not the legacy commands.

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Transaction.hs [hunk 1]
```diff
@@ -215,10 +216,20 @@ pTransactionBuildRaw era =
       <*> many pMetadataFile
       <*> optional pProtocolParamsFile
       <*> pFeatured era (optional pUpdateProposalFile)
-      <*> many (pFileInDirection "vote-file" "Filepath of the vote.")
-      <*> many (pFileInDirection "proposal-file" "Filepath of the proposal.")
+      <*> pVoteFile era
+      <*> pProposalFile era
       <*> pTxBodyFileOut
 
+pVoteFile :: ShelleyBasedEra era -> Parser [VoteFile In]
```
Comment (carlhammann) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/Transaction.hs]: Maybe use a plural name here, like `pVoteFiles`, as we're parsing zero or more?

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Transaction.hs [hunk 2]
```diff
@@ -215,10 +216,20 @@ pTransactionBuildRaw era =
...
+pVoteFile = caseShelleyToBabbageOrConwayEraOnwards 
+        (const $ pure [])
+        (const $ many (pFileInDirection "vote-file" "Filepath of the vote."))
+
+pProposalFile :: ShelleyBasedEra era -> Parser [ProposalFile In]
```
Comment (carlhammann) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/Transaction.hs]: Plural, as with the vote files?

**Review by carlhammann — APPROVED** [review]
Nicely done, and props for using `VoteFile` instead of `File`!

**Review by carbolymer — APPROVED** [review]

**Pull request merged**